src/openfermion/ops/_interaction_tensor.py

```diff
@@ -84,103 +85,114 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
+    Much of the functionality of this class is redudant with FermionOperator
+    but enables much more efficient numerical computations in many cases,
+    such as basis rotations.
 
     Attributes:
-        n_qubits: The number of qubits on which the tensor
+        n_qubits(int): The number of qubits on which the tensor
             acts.
-        constant: A constant term in the operator given as a float.
-            For instance, the nuclear repulsion energy.
-        one_body_tensor: The coefficients of the 2D matrix terms. This is an
-            n_qubits x n_qubits numpy array of floats. For instance, the one
-            body term of MolecularOperator.
-        two_body_tensor: The coefficients of the 4D matrix terms. This is an
-            n_qubits x n_qubits x n_qubits x n_qubits numpy array offloats.
-            For instance, the two body term of MolecularOperator.
+        constant(complex): A constant term in the operator given as a complex
+            number. For instance, the nuclear repulsion energy.
+        n_body_tensors(dict): A dictionary storing the tensors describing
+            n-body interactions. For instance, n_body_tensors[2] is a
+            n_qubits x n_qubits x n_qubits x n_qubits numpy array of complex
+            numbers.
     """
 
-    def __init__(self, constant, one_body_tensor, two_body_tensor):
+    def __init__(self, constant, *args, n_body_tensors=None, n_qubits=None):
         """Initialize the InteractionTensor class.
 
         Args:
-            constant: A constant term in the operator given as a
-                float. For instance, the nuclear repulsion energy.
-            one_body_tensor: The coefficients of the 2D matrix terms. This
-                is an n_qubits x n_qubits numpy array of floats. For
-                instance, the one body term of MolecularOperator.
-            two_body_tensor: The coefficients of the 4D matrix terms. This is
-                an n_qubits x n_qubits x n_qubits x n_qubits numpy array of
-                floats. For instance, the two body term of MolecularOperator.
+            constant(complex): A constant term in the operator given as a
+                complex number. For instance, the nuclear repulsion energy.
+            n_body_tensors(dict): A dictionary storing the tensors describing
+                n-body interactions. For instance, n_body_tensors[2] is a
+                n_qubits x n_qubits x n_qubits x n_qubits numpy array of
+                complex numbers.
         """
-        if constant is None:
-            constant = 0.0
-        self.constant = constant
-        self.one_body_tensor = one_body_tensor
-        self.two_body_tensor = two_body_tensor
-        self.n_qubits = self.one_body_tensor.shape[0]
+        # initialize constant
+        if constant is not None:
+            self.constant = constant
+        else:
+            self.constant = 0.
+
+        # initialize n_body_tensors
```

Comment:
I agree that it would be cleaner to have one method of initialization. I might be misunderstanding, but it sounds to me like you are describing initializing by passing in a FermionOperator (or equivalent information). I think that sort of thing should be relegated to conversion routines (as it is right now). I myself am partial to just keeping the initialization with **args, allowing the input to be an arbitrary list of tensors. Maybe @babbush has some thoughts on this?

Reply:
I think keeping the args input would be fine and dropping the ability to also pass in the dictionary.

Reply:
Well my thoughts on the *args initialization depend on some of the issues we are discussing above. If we want the ability to keep hole matrices for instance, how can you even use the *args? For instance, isn't it just going to assume that the args would be normal tensors and not hole tensors? On this point, I would think using a dictionary as the only initialization makes sense but I'm not sure.

I agree with Kevin that we should try to stick to the single-responsibility principle (will make things more compatible with C++ for instance later on) and use a conversion function (not method) to transform from FermionOperator.